### PR TITLE
Add containsKey and containsValue to make it consistent with findValue and findKey

### PR DESCRIPTION
**Pull request by tomasfejfar**
* added hasKey()
* added hasValue*()
* deprecated inArray*() and proxy to hasValue*()

### CODE REVIEW

File: src/Type/ArrayType/ArrayType.php [hunk 1]
```diff
@@ -422,4 +422,19 @@ public static function uniqueValuesByCallback(array $haystack, Closure $callback
 		return $result;
 	}
 
+	/**
+	 * Mimics array_key_exists
```
Comment (VasekPurchart) [inline src/Type/ArrayType/ArrayType.php]: Probably better `Wrapper for array_key_exists`?

File: src/Type/ArrayType/ArrayType.php [hunk 2]
```diff
@@ -422,4 +422,19 @@ public static function uniqueValuesByCallback(array $haystack, Closure $callback
...
+	 *
+	 * @param array $haystack
```
Comment (VasekPurchart) [inline src/Type/ArrayType/ArrayType.php]: `mixed[]`

File: src/Type/ArrayType/ArrayType.php [hunk 3]
```diff
@@ -422,4 +422,19 @@ public static function uniqueValuesByCallback(array $haystack, Closure $callback
...
+	 * @param integer|string $key
+	 * @return boolean key exists
```
Comment (VasekPurchart) [inline src/Type/ArrayType/ArrayType.php]: I think the description does not add anything?

File: src/Type/ArrayType/ArrayType.php [hunk 4]
```diff
@@ -422,4 +422,19 @@ public static function uniqueValuesByCallback(array $haystack, Closure $callback
...
+	 */
+	public static function hasKey(array $haystack, $key)
+	{
+		if (!array_key_exists($key, $haystack)) {
```
Comment (VasekPurchart) [inline src/Type/ArrayType/ArrayType.php]: Why not return the result directly?

File: tests/Type/ArrayType/ArrayTypeTest.php [hunk 5]
```diff
@@ -558,4 +558,17 @@ public function testUniqueValuesByCallbackWithNonStrictComparisonWithObjects()
 		$this->assertCount(1, $actual);
 	}
 
+	public function testHasKeyIsNotStrict()
```
Comment (VasekPurchart) [inline tests/Type/ArrayType/ArrayTypeTest.php]: Please add a test for a null value - becase it differs from `isset` for example https://3v4l.org/Kd0c6
And any other weird values you know of :)

File: src/Type/ArrayType/ArrayType.php [hunk 6]
```diff
@@ -22,10 +22,11 @@ class ArrayType extends \Consistence\ObjectPrototype
 	 * @param mixed $needle
 	 * @param boolean $strict
 	 * @return boolean
+	 * @deprecated
```
Comment (VasekPurchart) [inline src/Type/ArrayType/ArrayType.php]: @deprecated should be above the type annotations, see https://github.com/consistence/coding-standard/blob/master/consistence-coding-standard.md#phpdoc

File: src/Type/ArrayType/ArrayType.php (hunk 1 above)
Comment (tomasfejfar) [inline src/Type/ArrayType/ArrayType.php]: 👍

File: src/Type/ArrayType/ArrayType.php (hunk 3 above)
Comment (tomasfejfar) [inline src/Type/ArrayType/ArrayType.php]: :+1

File: src/Type/ArrayType/ArrayType.php (hunk 4 above)
Comment (tomasfejfar) [inline src/Type/ArrayType/ArrayType.php]: It should. It's residue from idea I had about testing whether the key is strictly the same - which does not make sense as array converts the values anyway :( (i.e. will make `1` from `"1"`)

File: tests/Type/ArrayType/ArrayTypeTest.php (hunk 5 above)
Comment (tomasfejfar) [inline tests/Type/ArrayType/ArrayTypeTest.php]: 👍

**Comment by tomasfejfar** [comment]
`hasValue`/`findValue`/`getValue`/etc. works better for me as I prefer pattern-based consistence. `contains` may be the correct term for what it does, but I imagine it would be harded for me to remember it on day-to-day basis. I'll leave the decision to you as I only have preference, not concrete proof of why one is better than the other.

File: tests/Type/ArrayType/ArrayTypeTest.php (hunk 5 above)
Comment (tomasfejfar) [inline tests/Type/ArrayType/ArrayTypeTest.php]: Done.

**Comment by tomasfejfar** [comment]
Regarding the "duplicate" tests, I prefer it this way as it explicitly tests each method of the API. Keeping only one of the methods does not seem right. The test would then rely on the "internal" knowledge that the old methods proxy to old ones. When the old methods are removed, so should the tests for them be removed. 

I will however sort the methods in code so that the same methods are near each other to better convey the message that they are alternative to the old ones.

File: src/Type/ArrayType/ArrayType.php [hunk 7]
```diff
@@ -18,38 +18,78 @@ class ArrayType extends \Consistence\ObjectPrototype
 	/**
 	 * Wrapper for PHP in_array, provides safer default parameter
 	 *
+	 * @deprecated
```
Comment (mhujer) [inline src/Type/ArrayType/ArrayType.php]: What about a message? `use ArrayType::hasValue() instead` ? (same for the others)

Comment (VasekPurchart) [inline src/Type/ArrayType/ArrayType.php]: Thanks a good idea :)

**Comment by VasekPurchart** [comment]
I have:

* fixed the remaining issues
* rearranged the test methods so that they match the original ones
* added the test case I mentioned in https://3v4l.org/Kd0c6
* documented new methods in old methods @deprecated annotations
* used `contains` because I have discuessed this with several other people and it seems to suit more, also is widely used, e.g.:
  * http://www.doctrine-project.org/api/common/2.3/class-Doctrine.Common.Collections.Collection.html#_contains
  * https://docs.oracle.com/javase/7/docs/api/java/util/Collection.html#contains(java.lang.Object)
  * https://docs.oracle.com/javase/7/docs/api/java/util/Map.html#containsKey(java.lang.Object)
  * https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html#contains(java.lang.Object)
  * other uses alreday in Consistence and Sentry
* added new method `containsKeyByValueCallback` to be consistent with all of the other methods
* changed `constainsValueByCallback` to only `constainsByCallback` since it combines searching by both key and value

**Comment by VasekPurchart** [comment]
And changed all the places, where the originals were used.
